codegen/grid.py

```diff
@@ -0,0 +1,665 @@
+from sympy import Symbol, symbols, factorial, Matrix, Rational, Indexed, simplify, IndexedBase, solve, Eq
+from mako.template import Template
+from mako.lookup import TemplateLookup
+from mako.runtime import Context
+from StringIO import StringIO
+
+from sympy.printing.ccode import CCodePrinter
+
+hf = Rational(1,2) # 1/2
+
+
+class MyCPrinter(CCodePrinter):
+	def __init__(self, settings={}):
+		CCodePrinter.__init__(self, settings)
+
+	def _print_Indexed(self, expr):
+		# Array base and append indices
+		output = self._print(expr.base.label) + ''.join(['[' + self._print(x) + ']' for x in expr.indices])
+		return output
+
+	def _print_Rational(self, expr):
+		p, q = int(expr.p), int(expr.q)
+		return '%d.0F/%d.0F' % (p, q) # float precision by default
+
+	def _print_Mod(self, expr):
+		# print mod as % in C++
+		args = map(ccode, expr.args)
+		args = ['('+x+')' for x in args]
+		result = '%'.join(args)
+		return result
+
+def ccode(expr, **settings):
+	return MyCPrinter(settings).doprint(expr, None)
+
+def ccode_eq(eq, **settings):
+	# print assignment code from equation
+	return MyCPrinter(settings).doprint(eq.lhs, None) + ' = ' + MyCPrinter(settings).doprint(eq.rhs, None)
+
+def render(tmpl, dict1):
+	buf = StringIO()
+	ctx = Context(buf, **dict1)
+	tmpl.render_context(ctx)
+	return buf.getvalue()
+
+def IndexedBases(s):
+	l = s.split();
+	bases = [IndexedBase(x) for x in l]
+	return tuple(bases)
+
+def tc(dx, n):
+    # return coefficient of power n Taylor series term
```

Comment:
These comment lines should really be """docstrings""" instead. Same applies to the other functions below.